includes/class-newspack-popups-model.php

```diff
@@ -240,31 +478,108 @@ public static function retrieve_preview_popup( $post_id ) {
+				'large_border'                   => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['large_border'], FILTER_VALIDATE_BOOLEAN ),
+				'frequency'                      => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['frequency'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'frequency_max'                  => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['frequency_max'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'frequency_start'                => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['frequency_start'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'frequency_between'              => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['frequency_between'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'frequency_reset'                => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['frequency_reset'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'overlay_color'                  => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['overlay_color'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'overlay_opacity'                => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['overlay_opacity'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'overlay_size'                   => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['overlay_size'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'no_overlay_background'          => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['no_overlay_background'], FILTER_VALIDATE_BOOLEAN ),
+				'placement'                      => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['placement'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'trigger_type'                   => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['trigger_type'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'trigger_delay'                  => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['trigger_delay'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'trigger_scroll_progress'        => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['trigger_scroll_progress'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'trigger_blocks_count'           => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['trigger_blocks_count'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'archive_insertion_posts_count'  => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['archive_insertion_posts_count'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
+				'archive_insertion_is_repeating' => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['archive_insertion_is_repeating'], FILTER_VALIDATE_BOOLEAN ),
+				'utm_suppression'                => filter_input( INPUT_GET, Newspack_Popups::PREVIEW_QUERY_KEYS['utm_suppression'], FILTER_SANITIZE_FULL_SPECIAL_CHARS ),
```

Comment:
Same as per the previous comment, could improve maintainability by reducing code repetition. Maybe a separate method that handles both cases?

Reply:
Good idea—implemented in 49cdc86.